compiler/parser.nim

```diff
@@ -1795,6 +1795,14 @@ proc parseRoutine(p: var Parser, kind: TNodeKind): PNode =
   else:
     result.add(p.emptyNode)
   indAndComment(p, result, maybeMissEquals)
+  let body = result[^1]
+  if body.kind == nkStmtList and body.len > 0 and body[0].comment.len > 0 and body[0].kind != nkCommentStmt:
+    assert result.comment.len == 0
```

Comment:
Better make this assert part of the `if` condition instead. Don't want this tiny change to break anybody's code.

Reply:
For core compiler code all assertions should be `doAssert`,
otherwise you may have hard to detect different behaviours across CI depending on certain flags, or not.

Reply:
> Better make this assert part of the if condition instead. Don't want this tiny change to break anybody's code.

done for sake of moving on, but this is not ideal, because we'll never know if that condition is violated; I can see why the assert could be a problem if it fails in a nim release version; 

in future work we should do this:
* add a new warning `warnInternalError` and turn it on by default
* issue it in cases like this

> For core compiler code all assertions should be doAssert,

nim now builds with -d:release, which keeps assertions on; it'd only make a difference if a user builds with -d:danger; although in this case, the surrounding cost dominates so it's not an issue either way; in any case there is no more assert now